Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst

```diff
@@ -1,298 +1,336 @@
+:sql:`JOIN ON` conditions. Functions like :php:`->min()` may also be used in
+:sql:`SELECT` parts.
 
-.. code-block:: php
-   :caption: EXT:some_extension/Classes/SomeClass.php
+It takes care of building query conditions and ensures that table and column
+names are quoted within the created expressions / SQL fragments. It is a facade
+to the actual Doctrine DBAL :php:`ExpressionBuilder`.
 
-   $expressionBuilder = $queryBuilder->expr();
+The expression builder is used in the context of the :ref:`query builder
+<database-query-builder>` to ensure that queries are built based on the
+requirements of the database platform being used.
 
-It is good practice to not assign an instance of the `ExpressionBuilder` to a variable but
-to use it within the code flow of the `QueryBuilder` context directly:
 
-.. code-block:: php
-   :caption: EXT:some_extension/Classes/SomeClass.php
-
-   // use TYPO3\CMS\Core\Utility\GeneralUtility;
-   // use TYPO3\CMS\Core\Database\ConnectionPool;
-   $rows = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content')
-      ->select('uid', 'header', 'bodytext')
-      ->from('tt_content')
-      ->where(
-         // `bodytext` = 'klaus' AND `header` = 'peter'
-         $queryBuilder->expr()->eq('bodytext', $queryBuilder->createNamedParameter('klaus')),
-         $queryBuilder->expr()->eq('header', $queryBuilder->createNamedParameter('peter'))
-      )
-      ->executeQuery()
-      ->fetchAllAssociative();
-
-.. warning::
+Basic usage
+===========
 
-   It is crucially important to quote values correctly to not introduce SQL injection attack
-   vectors to your application. See the
-   :ref:`section of the QueryBuilder <database-query-builder-create-named-parameter>` for details.
+An instance of the :php:`ExpressionBuilder` is retrieved from the
+:php:`QueryBuilder` object:
 
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
 
-Junctions
-=========
+    $expressionBuilder = $queryBuilder->expr();
 
-.. versionchanged:: 11.5.10
-   The methods :php:`andX()` and :php:`orX()` are deprecated and replaced
-   by :php:`and()` and :php:`or()` to match with `doctrine/dbal`, which
-   `deprecated <https://github.com/doctrine/dbal/commit/84328cd947706210caebcaea3ca0394b3ebc4673>`_
-   these methods.
+It is good practice not to assign an instance of the :php:`ExpressionBuilder` to
+a variable, but to use it directly within the code flow of the query builder
+context:
 
-* :php:`->and()` conjunction
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
 
-* :php:`->or()` disjunction
+    $rows = $this->connectionPool->getQueryBuilderForTable('tt_content')
+        ->select('uid', 'header', 'bodytext')
+        ->from('tt_content')
+        ->where(
+            // `bodytext` = 'klaus' AND `header` = 'peter'
+            $queryBuilder->expr()->eq('bodytext', $queryBuilder->createNamedParameter('klaus')),
```

Comment:
Where does `$queryBuilder` come from?

Reply:
Yup, you are right, that is inconsistent.